lib/Segments/DynamicSegments/FilterDataMapper.php

```diff
@@ -42,6 +39,26 @@ private function getSegmentType(array $data): string {
     return $data['segmentType'];
   }
 
+  private function createSubscriber(array $data): DynamicSegmentFilterData {
+    if (empty($data['action'])) {
+      $data['action'] = DynamicSegmentFilterData::TYPE_USER_ROLE;
+    }
+    if ($data['action'] === SubscriberSubscribedDate::TYPE) {
+      if (empty($data['value'])) throw new InvalidFilterException('Missing number of days', InvalidFilterException::MISSING_VALUE);
+      return new DynamicSegmentFilterData([
+        'segmentType' => DynamicSegmentFilterData::TYPE_USER_ROLE,
+        'action' => $data['action'],
+        'value' => $data['value'],
+        'operator' => $data['operator'] ?? 'before',
+      ]);
+    }
+    if (empty($data['wordpressRole'])) throw new InvalidFilterException('Missing role', InvalidFilterException::MISSING_ROLE);
+    return new DynamicSegmentFilterData([
+      'segmentType' => DynamicSegmentFilterData::TYPE_USER_ROLE,
+      'wordpressRole' => $data['wordpressRole'],
+    ]);
+  }
```

Comment:
Since the logic for mapping use role type segment became more complex I guess it would be nice to add a few test cases in related unit test https://github.com/mailpoet/mailpoet/blob/master/tests/unit/Segments/DynamicSegments/FilterDataMapperTest.php